bob-access/src/extractor.rs

```diff
@@ -40,14 +53,16 @@ impl<T: Extractor> ExtractorExt for T {
 
     fn extract_basic(&self) -> Result<Credentials, Error> {
         let mut builder = prepare_builder(self)?;
-        match (self.get_header("username")?, self.get_header("password")?) {
-            (Some(username), Some(password)) => {
+        let auth_header = self.get_header("authorization")?;
+        if let Some(auth_header) = auth_header {
+            let mut parts = auth_header.split_whitespace();
+            if let (Some("Basic"), Some(credentials)) = (parts.next(), parts.next()) {
+                let (username, password) = username_password_from_credentials(credentials)?;
                 let creds = builder
                     .with_username_password(username, password)
                     .build();
-                Ok(creds)
-            },
-            (None, None) => {
+                return Ok(creds);
+            } else {
                 if let Some(node_name) = self.get_header("node_name")? {
```

Comment:
I think it would be good to perform inter-node auth also through `authorization` header. We can use similar format: `Authorization: InterNode <base64 node_name>`